CMakeLists.txt

```diff
@@ -12,8 +12,13 @@ elseif ("${CMAKE_GENERATOR}" STREQUAL "Visual Studio 14 2015 Win64")
   set(CMAKE_GENERATOR_TOOLSET "v140</PlatformToolset></PropertyGroup><ItemDefinitionGroup Condition=\"'$(ProjectName)'=='folly'\"><ProjectReference><LinkLibraryDependencies>true</LinkLibraryDependencies></ProjectReference></ItemDefinitionGroup><PropertyGroup><PlatformToolset>v140")
   set(MSVC_IS_2017 OFF)
 elseif ("${CMAKE_GENERATOR}" STREQUAL "Ninja")
-  message("Folly is being built with Ninja, so assuming VS 2017 is being used.")
-  set(MSVC_IS_2017 ON)
+  if (MSVC_VERSION EQUAL 1910)
```

Comment:
Unfortunately, no, as you can't change the generator toolset after the compiler has been tested :( Once I get the reason #589 was needed resolved, it should be possible to move this entire check into `FollyCompiler.cmake`.